doc/users/installing/index.rst

```diff
@@ -92,7 +92,12 @@ as the package index to query:
 
 .. code-block:: sh
 
-  python -m pip install --upgrade --index-url https://pypi.anaconda.org/scipy-wheels-nightly/simple matplotlib
+  python -m pip install \
+    --upgrade \
+    --pre \
```

Comment:
Hm. The one downside to trying to do this is one command instead of something like

```
python -m pip install --upgrade matplotlib  # Get all dependencies
python -m pip install --upgrade --index-url https://pypi.anaconda.org/scipy-wheels-nightly/simple matplotlib  # Get the nightly
```

is that `--pre` is applied universally, and so this will also get pre releases of `matplotlib`'s dependencies from PyPI, which might not be the desired effect. Of course on the flip side of this

```
python -m pip install --upgrade --index-url https://pypi.anaconda.org/scipy-wheels-nightly/simple matplotlib
```

doesn't know about PyPI at all and so if `matplotlib` changed its dependencies in the nightly (like in PR #22751) then this wouldn't be able to get picked up properly.

My assumption is that the use of `--pre` is probably worth the risk here to make things easier in general. Though if people think we should add an admonition to warn people I'm happy to add something like that.

Reply:
I would err on side of this command "just working" in the naive case.  I think that splitting this up by lines is pretty good because it is very clear which pip flags are being called.  

If people are signing up for nightlies then I think they are agreeing to debug any issues they run into and on the scale of things that can go wrong, accidentally pulling pre-releases is pretty low on the list 😉 .  It might even be a feature as it is getting more future version <-> future version compatibility testing done .

Reply:
In addition I think that this is the only correct way (short of manually parsing mpl's setup.py) to get the dependencies that will work. e.g. when mpl adds contourpy as a dep then the two step install will break

Reply:
> If people are signing up for nightlies then I think they are agreeing to debug any issues they run into and on the scale of things that can go wrong, accidentally pulling pre-releases is pretty low on the list  . It might even be a feature as it is getting more future version <-> future version compatibility testing done .

This is the same conclusion I came to after eating and making coffee, so SGTM!. :)